src/layout.rs

```diff
@@ -318,6 +318,21 @@ fn split(area: Rect, layout: &Layout) -> Rc<[Rect]> {
             });
         }
     }
+    for (i, c1) in elements.iter().enumerate() {
```

Comment:
I actually like keeping all the constraints in the same function. What do you think about me adding comments instead?

Reply:
My general rationale for adding a function instead of just commented sections is that everything you need to understand in a particular block is limited to the scope of the function. A commented section could be affected by any change above, and any change within can effect any code below, which means you have to understand the full function to maintain it. In any particular part of the lib, that's not a big deal, but when large functions dominate, every maintenance task tends to become larger as the time spent understanding is magnified.

Reply:
Fair. In that case I think I’d like to move all the different constraints into individual functions too then.

Reply:
With the recent refactors, I'm assuming it is okay to have the equations in the same function.